services/core-api/app/api/securities/resources/bond.py

```diff
@@ -98,8 +99,11 @@ def put(self, bond_guid):
 
```

Comment:
The catch of this block should delete the bond history object too

Reply:
it does

Reply:
![image](https://user-images.githubusercontent.com/17113094/84418805-8918bf80-abcc-11ea-9411-d8e0d427e8f6.png)
We need to do `history.delete()` between lines 97 and 98 too.

Reply:
I would also raise a specific error on 106, maybe InternalServerError with the exception message?

Reply:
The second exception block around the save already raises a SQLAlchemy exception to I am just catching it again so I can delete the history and then re-raising the initial exception.